Formula/a/azure-cli.rb

```diff
@@ -594,6 +595,11 @@ class AzureCli < Formula
     sha256 "aac08f26a31dc4dffd92821527d1682d99d52f9ef6851968114a8728f3c274d3"
   end
 
+  resource "pip" do
+    url "https://files.pythonhosted.org/packages/4d/87/fb90046e096a03aeab235e139436b3fe804cdd447ed2093b0d70eba3f7f8/pip-24.2.tar.gz"
+    sha256 "5b5e490b5e9cb275c879595064adce9ebd31b854e3e803740b72f9ccf34a45b8"
+  end
```

Comment:
We don't need a `pip` resource, just do
```diff
diff --git a/Formula/a/azure-cli.rb b/Formula/a/azure-cli.rb
index c09f30b050a..29f819663e7 100644
--- a/Formula/a/azure-cli.rb
+++ b/Formula/a/azure-cli.rb
@@ -719,7 +719,7 @@ class AzureCli < Formula
     ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
     ENV["OPENSSL_NO_VENDOR"] = "1"
 
-    venv = virtualenv_create(libexec, "python3.12", system_site_packages: false)
+    venv = virtualenv_create(libexec, "python3.12", system_site_packages: false, without_pip: false)
     venv.pip_install resources
 
     # Get the CLI components we'll install
```

Reply:
That's how we had it for Python 3.11, but that's no longer supported:
```
ArgumentError: virtualenv_create's without_pip is deprecated starting with Python 3.12
```

Reply:
Hm, weird.